File: 0995-minimum-number-of-k-consecutive-bit-flips/0995-minimum-number-of-k-consecutive-bit-flips.py
class Solution:
    def minKBitFlips(self, nums: List[int], k: int) -> int:
        ans = 0
        q = []
        for i in range(len(nums)):
            if len(q) % 2 == 0:
                if nums[i] == 0:
                    if i+k-1 <= len(nums)-1:
                        ans += 1
                        q.append(i+k-1)
                    else:
                        return -1
            else:
                if nums[i] == 1:
                    if i+k-1 <= len(nums)-1:
                        ans += 1
                        q.append(i+k-1)
                    else:
                        return -1
            if q:
                if q[0] == i:
                    q.pop(0)
        return ans

File: 0995-minimum-number-of-k-consecutive-bit-flips/0995-minimum-number-of-k-consecutive-bit-flips.cpp
class Solution {
public:
    int minKBitFlips(vector<int>& nums, int k) {
        int ans = 0;

        int i = 0, j = 0;
        int n = nums.size();
        vector<int> reset(n,0);
        
        int flag = 1;
        while(i<=(n-k)){
            flag = flag^reset[i];
            if(nums[i]^flag){
                ans++;
                if(i+k<n){
                    reset[i+k] = 1;
                }
                flag = !flag;
            }
            i++;
        }

        while(i<n){
            flag = flag^reset[i];
            if(nums[i]^flag){
                return -1;
            }
            i++;
        }

        return ans;
    }
};